include/boost/geometry/extensions/multi/gis/io/wkb/detail/parser.hpp

```diff
@@ -0,0 +1,237 @@
+            
+            detail::wkb::byte_order_type::enum_t linestring_byte_order;
+            if (!detail::wkb::byte_order_parser::parse(it, end, linestring_byte_order))
+            {
+                return false;
+            }
+            
+            if (!geometry_type_parser<linestring_type>::parse(it, end, order))
+            {
+                return false;
+            }
+            
+            if(!point_container_parser<linestring_type>::parse(it, end, linestring_buffer, linestring_byte_order))
+            {
+                return false;
+            }
+            
+            output = linestring_buffer;
+            
+            ++output;
+            ++linestrings_parsed;
+        }
+        return true;
+    }
+};
+
+template <typename MultiPolygon>
+struct multipolygon_parser
+{
+    template <typename Iterator>
+    static bool parse(Iterator& it, Iterator end, MultiPolygon& multipolygon, byte_order_type::enum_t order)
+    {
+        typedef typename MultiPolygon::value_type polygon_type;
+        typedef typename ring_type<MultiPolygon>::type ring_type;
+        
+        if (!geometry_type_parser<MultiPolygon>::parse(it, end, order))
+        {
+            return false;
+        }
+        
+        boost::uint32_t num_polygons(0);
+        if (!value_parser<boost::uint32_t>::parse(it, end, num_polygons, order))
+        {
+            return false;
+        }
+        
+        std::back_insert_iterator<MultiPolygon> output(std::back_inserter(multipolygon));
+        
+        std::size_t polygons_parsed = 0;
+        while(polygons_parsed < num_polygons && it != end)
+        {
+            polygon_type polygon_buffer;
+            
+            detail::wkb::byte_order_type::enum_t polygon_byte_order;
+            if (!detail::wkb::byte_order_parser::parse(it, end, polygon_byte_order))
+            {
+                return false;
+            }
+            
+            if (!geometry_type_parser<polygon_type>::parse(it, end, order))
+            {
+                return false;
+            }
+            
+            boost::uint32_t num_rings(0);
+            if (!value_parser<boost::uint32_t>::parse(it, end, num_rings, polygon_byte_order))
+            {
+                return false;
+            }
+            
+            std::size_t rings_parsed = 0;
+            
+            while (rings_parsed < num_rings && it != end)
+            {
+                
+                if (0 == rings_parsed)
+                {
+                    ring_type& ring0 = exterior_ring(polygon_buffer);
+                    
+                    if (!point_container_parser<ring_type>::parse(it, end, ring0, polygon_byte_order))
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    interior_rings(polygon_buffer).resize(rings_parsed);
+                    ring_type& ringN = interior_rings(polygon_buffer).back();
```

Comment:
Should be `range::resize()` and `range::back()` and instead of `ring_type&` `range_reference<...>::type`.